src/status_im/protocol/web3/inbox.cljs

```diff
@@ -1,67 +1,79 @@
 (def default-enode "enode://0f51d75c9469de0852571c4618fe151265d4930ea35f968eb1a12e69c12f7cbabed856a12b31268a825ca2c9bafa47ef665b1b17be1ab71de83338c4b7439b24@127.0.0.1:30303")
 
-;; adamb's status-cluster enode
-(def cluster-enode (:address constants/default-wnode))
+;; NOTE(dmitryn) Expects JSON response like:
+;; {"error": "msg"} or {"result": true}
+(defn- parse-json [s]
+  (try
+    (-> s
+        js/JSON.parse
+        (js->clj :keywordize-keys true))
+    (catch :default e
+      {:error (.-message e)})))
 
-;; TODO(oskarth): Rewrite callback-heavy code with CSP and/or coeffects
-;; TODO(oskarth): Memoize addPeer and markTrusted, similar to keys/get-sym-key
-;; TODO(oskarth): Actually deal with errors, all in same cb - outside scope of this
-(defn request-messages! [web3 {:keys [enode topic password]
-                               :or {enode    default-enode
-                                    password inbox-password
-                                    topic    default-topic}} callback]
-  (status/add-peer
-   enode
-   (fn [res]
-     (log/info "offline inbox: add peer" enode res)
-     (let [args {:jsonrpc "2.0"
-                 :id      1
-                 :method  "shh_markTrustedPeer"
-                 :params  [enode]}
-           payload (.stringify js/JSON (clj->js args))]
-       (log/info "offline inbox: mark-trusted-peer request")
-       (status/call-web3
-        payload
-        (fn [res2]
-          (log/info "offline inbox: mark-trusted-peer response" enode res2)
-          (keys/get-sym-key web3 password
-                            (fn [sym-key-id]
-                              (log/info "offline inbox: sym-key-id" sym-key-id)
-                              (let [args {:jsonrpc "2.0"
-                                          :id      2
-                                          :method  "shh_requestMessages"
-                                          ;; NOTE: "from" and "to" parameters omitted here
-                                          ;; by default "from" is 24 hours ago and "to" is time now
-                                          :params  [{:mailServerPeer enode
-                                                     :topic          topic
-                                                     :symKeyID       sym-key-id}]}
-                                    payload (.stringify js/JSON (clj->js args))]
-                                (log/info "offline inbox: request-messages request")
-                                (log/info "offline inbox: request-messages args" (pr-str args))
-                                (log/info "offline inbox: request-messages payload" (pr-str payload))
+(defn- response-handler [error-fn success-fns]
+  (fn [response]
+    (let [{:keys [error result]} (parse-json response)]
+      ;; NOTE(dmitryn): AddPeer() may return {"error": ""}
+      ;; assuming empty error is a success response
+      (if (or (nil? error)
```

Comment:
Not 100% what string/blank? does but perhaps this can be replaced by `if seq`?

Reply:
changed to `if seq`. actually `string/blank?` already checks for nil, so `nil?` check was really redundant there.